### PR TITLE
Facebook login

### PR DESCRIPTION
**Pull request by AndrewSakovich**

### CODE REVIEW

File: src/redux/saga/signOutSaga.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import auth from '@react-native-firebase/auth';
-import { GoogleSignin } from '@react-native-google-signin/google-signin';
 import { signOutAction } from '../actions/authActions/signOutAction';
 import { put } from 'redux-saga/effects';
 import { createErrorAlertMessageHelper } from '../../helpers/createErrorAlertMessageHelper';
 
 export function* signOutSaga() {
   try {
     yield auth().signOut();
-    yield GoogleSignin.revokeAccess();
```
Comment (SHEMPEL133) [inline src/redux/saga/signOutSaga.ts:10]: you don't need it anymore?

File: src/screens/LoginScreen/index.tsx [hunk 2]
```diff
@@ -3,18 +3,28 @@ import { View } from 'react-native';
...
 
 export const LoginScreen: FC = () => {
   const dispatch = useDispatch();
 
-  const googleSignIn = () => {
-    dispatch(signInSagaAction());
+  const googleObject = {
+    signInMethod: () => dispatch(googleSignInSagaAction()),
+    title: 'Sign in with Google',
+    typeStyle: 'google',
```
Comment (SHEMPEL133) [inline src/screens/LoginScreen/index.tsx]: if you provide type, make sense made all parameters configurable by type 

also typy should be enum

File: src/redux/saga/fbSignInSaga.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  try {
+    // Attempt login with permissions
+    yield LoginManager.logInWithPermissions(['public_profile', 'email']);
+
+    // Once signed in, get the users AccessToken
+    const dataToken: AccessTokenMap = yield AccessToken.getCurrentAccessToken();
+    const accessToken = dataToken.accessToken;
+    // const user = profileRequest.callback();
+    if (!accessToken) {
+      createErrorAlertMessageHelper(
```
Comment (SHEMPEL133) [inline src/redux/saga/fbSignInSaga.ts]: code below `'if'` should be executed after alert?

File: src/redux/saga/fbSignInSaga.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        'Something went wrong obtaining access accessToken',
+      );
+    }
+
+    console.log('dataToken', dataToken);
+    // Create a Firebase credential with the AccessToken
+    const facebookCredential = auth.FacebookAuthProvider.credential(
+      dataToken.accessToken,
+    );
+    console.log('facebookCredential   ', facebookCredential);
```
Comment (SHEMPEL133) [inline src/redux/saga/fbSignInSaga.ts]: remove console log please

File: src/redux/actions/authSagaActions/index.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 export enum AuthSagaActions {
-  SIGN_IN_SAGA = 'SIGN_IN_SAGA',
+  GOOGLE_SIGN_IN_SAGA = 'GOOGLE_SIGN_IN_SAGA',
+  FB_SIGN_IN_SAGA = 'FB_SIGN_IN_SAGA',
```
Comment (SHEMPEL133) [inline src/redux/actions/authSagaActions/index.ts]: write facebook, not `fb`

File: android/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -1,25 +1,27 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-  package="com.appwithsignin">
+          package="com.appwithsignin">
```
Comment (evgenyshenets91) [inline android/app/src/main/AndroidManifest.xml]: What does this change mean? Please every time check your PR from the top to bottom. I suppose that is not necessary

File: src/components/SignInButton/index.tsx [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
-import { StatusBar, Text, TouchableOpacity, View } from 'react-native';
+import { Text, TouchableOpacity } from 'react-native';
 import { SignInButtonPropsType } from './type';
+import { style } from './style';
 
 export const SignInButton: FC<SignInButtonPropsType> = props => {
-  const { googleSignIn } = props;
+  const { signIn } = props;
+
+  const textStyle = () => {
```
Comment (evgenyshenets91) [inline src/components/SignInButton/index.tsx]: make it like a variable, not function. and apply cond rendering

File: src/redux/saga/fbSignInSaga.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    const dataToken: AccessTokenMap = yield AccessToken.getCurrentAccessToken();
+    const accessToken = dataToken.accessToken;
+    // const user = profileRequest.callback();
+    if (!accessToken) {
+      createErrorAlertMessageHelper(
+        'Something went wrong obtaining access accessToken',
+      );
+    }
+
+    console.log('dataToken', dataToken);
```
Comment (evgenyshenets91) [inline src/redux/saga/fbSignInSaga.ts]: console.log! please be attentive

File: src/redux/saga/fbSignInSaga.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    // Create a Firebase credential with the AccessToken
+    const facebookCredential = auth.FacebookAuthProvider.credential(
+      dataToken.accessToken,
+    );
+    console.log('facebookCredential   ', facebookCredential);
+
+    // Sign-in the user with the credential
```
Comment (evgenyshenets91) [inline src/redux/saga/fbSignInSaga.ts]: I saw a similar code in google sign in, try to do a separate saga that will get params and will save the token and other necessary things, and will be called here and in google, because right now it looks like copy past from google.

File: src/redux/saga/signOutSaga.ts (hunk 1 above)
Comment (AndrewSakovich) [reply src/redux/saga/signOutSaga.ts:10]: yes, auth.SignOut will do it

File: src/redux/saga/fbSignInSaga.ts (hunk 3 above)
Comment (AndrewSakovich) [reply src/redux/saga/fbSignInSaga.ts]: i replase this to try/catch

File: src/redux/saga/fbSignInSaga.ts (hunk 4 above)
Comment (AndrewSakovich) [reply src/redux/saga/fbSignInSaga.ts]: ok

File: src/redux/actions/authSagaActions/index.ts (hunk 5 above)
Comment (AndrewSakovich) [reply src/redux/actions/authSagaActions/index.ts]: i fix  it

File: android/app/src/main/AndroidManifest.xml (hunk 6 above)
Comment (AndrewSakovich) [reply android/app/src/main/AndroidManifest.xml]: I'm sorry, I'll be more careful

File: src/components/SignInButton/index.tsx (hunk 7 above)
Comment (AndrewSakovich) [reply src/components/SignInButton/index.tsx]: i fix it

File: src/redux/saga/fbSignInSaga.ts (hunk 9 above)
Comment (AndrewSakovich) [reply src/redux/saga/fbSignInSaga.ts]: doing it

File: android/app/build.gradle [hunk 10]
```diff
@@ -194,6 +196,9 @@ dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
+implementation 'com.facebook.android:facebook-android-sdk:[8,9)'
```
Comment (evgenyshenets91) [inline android/app/build.gradle]: the same. add tabulation

File: android/app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -1,3 +1,5 @@
 <resources>
     <string name="app_name">Todo List</string>
+    <string name="facebook_app_id">1035486290709028</string>
+
```
Comment (evgenyshenets91) [inline android/app/src/main/res/values/strings.xml]: why did you add here empty line?

File: src/screens/LoginScreen/style.ts [hunk 12]
```diff
@@ -1,13 +1,43 @@
 import { StyleSheet, ViewStyle } from 'react-native';
+import { COLORS } from '../../COLORS';
 
-type LoginScreenStylesType = {
+export type LoginScreenStylesType = {
   container: ViewStyle;
+  googleButton: ViewStyle;
+  facebookButton: ViewStyle;
+  fontFacebook: ViewStyle;
```
Comment (evgenyshenets91) [inline src/screens/LoginScreen/style.ts]: it couldn't be viewstyle, it is textStyle and next style the same

File: src/redux/saga/googleSignInSaga.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+export function* googleSignInSaga() {
+  try {
+    const { idToken } = yield GoogleSignin.signIn();
+    // Create a Google credential with the accessToken
+    const googleCredential = auth.GoogleAuthProvider.credential(idToken);
+    // Sign-in the user with the credential
+    const { user } = yield auth().signInWithCredential(googleCredential);
+
+    const userToken = user.uid;
+    yield put(checkFirebaseUsersSagaAction({ user, userToken }));
```
Comment (evgenyshenets91) [inline src/redux/saga/googleSignInSaga.ts]: why did you call the action with the word "firebase" and use it in google? The name should be basic for both, like checkUserSagaAction

File: src/redux/saga/facebookSignInSaga.ts [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    const facebookCredential =
+      auth.FacebookAuthProvider.credential(accessToken);
+
+    // Sign-in the user with the credential
+    yield auth().signInWithCredential(facebookCredential);
+    const { user } = yield auth().signInWithCredential(facebookCredential);
+
+    const userToken = user.uid;
+
+    yield put(checkFirebaseUsersSagaAction({ user, userToken }));
```
Comment (evgenyshenets91) [inline src/redux/saga/facebookSignInSaga.ts]: It is not action creator. It is a saga as I understood. Use call effect instead of put

File: src/components/SignInButton/type.ts [hunk 15]
```diff
@@ -1,3 +1,18 @@
...
+import { IconDefinition } from '@fortawesome/fontawesome-common-types';
+
 export type SignInButtonPropsType = {
-  googleSignIn(): void;
+  signIn: SignInType;
+};
+type SignInType = {
+  signInMethod(): void;
+  title: string;
+  styleFont:
```
Comment (evgenyshenets91) [inline src/components/SignInButton/type.ts]: try to use here just TextStyle and further the same, it have to work

File: src/redux/saga/checkFirebaseUsersSaga.ts [hunk 16]
```diff
@@ -0,0 +1,26 @@
+import { FirebaseDatabaseTypes } from '@react-native-firebase/database';
+import { createReferenceHelper } from '../../helpers/createReferenceHelper';
+import { put } from 'redux-saga/effects';
+import { successSignInAction } from '../actions/authActions/successSignInAction';
+import { CheckFirebaseUsersSagaAction } from '../actions/checkFirebaseUsersSagaAction';
+
+export function* checkFirebaseUsersSaga(action: CheckFirebaseUsersSagaAction) {
+  const { userToken, user } = action.payload;
+  const usersData: FirebaseDatabaseTypes.DataSnapshot =
```
Comment (SHEMPEL133) [inline src/redux/saga/checkFirebaseUsersSaga.ts]: wrap your asyc code in try/catch

File: src/screens/LoginScreen/index.tsx [hunk 17]
```diff
@@ -3,18 +3,37 @@ import { View } from 'react-native';
...
+    icon: faFacebook,
+    iconColor: COLORS.sanMarino,
   };
 
   return (
     <View style={style.container}>
-      <SignInButton googleSignIn={googleSignIn} />
+      <SignInButton signIn={googleObject} />
+      <SignInButton signIn={facebookObject} />
+      {/*</View>*/}
```
Comment (SHEMPEL133) [inline src/screens/LoginScreen/index.tsx]: ?

File: src/components/SignInButton/style.ts [hunk 18]
```diff
@@ -1,27 +1,22 @@
...
-    flex: 1,
-    justifyContent: 'center',
-    alignItems: 'center',
-  },
-  btn: {
-    padding: 8,
-    borderWidth: 1,
-    borderRadius: 8,
-    borderColor: COLORS.sapphire,
+  fontGoogle: {
```
Comment (SHEMPEL133) [inline src/components/SignInButton/style.ts]: you don't use it and `fontFacebook`

File: src/components/SignInButton/index.tsx [hunk 19]
```diff
@@ -1,17 +1,17 @@
 import React, { FC } from 'react';
-import { style } from './style';
-import { StatusBar, Text, TouchableOpacity, View } from 'react-native';
+import { Text, TouchableOpacity } from 'react-native';
 import { SignInButtonPropsType } from './type';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
 
 export const SignInButton: FC<SignInButtonPropsType> = props => {
-  const { googleSignIn } = props;
+  const { signIn } = props;
```
Comment (SHEMPEL133) [inline src/components/SignInButton/index.tsx]: hardly understand what is `signIn` by this name

File: src/redux/saga/facebookSignInSaga.ts (hunk 14 above)
Comment (SHEMPEL133) [reply src/redux/saga/facebookSignInSaga.ts]: this is action creator
but @AndrewSakovich, make sense just call saga instead of dispatch action that is not handled in reducer and is not called from components

File: src/screens/LoginScreen/style.ts [hunk 20]
```diff
@@ -1,13 +1,43 @@
...
     flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  googleButton: {
+    width: '100%',
+    backgroundColor: COLORS.dawnPink,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
```
Comment (SHEMPEL133) [inline src/screens/LoginScreen/style.ts:23]: your style for google and facebook differs only in color 

I see two solutions
1) put this style in your button component as default and add your additional style from props 
2) in this style file create `const `with common styles and put for `googleButton` and 'facebookButton' using destructuring

the same for fonts style

File: src/screens/LoginScreen/index.tsx (hunk 17 above)
Comment (evgenyshenets91) [reply src/screens/LoginScreen/index.tsx]: lol

File: src/redux/saga/facebookSignInSaga.ts (hunk 14 above)
Comment (AndrewSakovich) [reply src/redux/saga/facebookSignInSaga.ts]: corrected

File: src/components/SignInButton/type.ts (hunk 15 above)
Comment (AndrewSakovich) [reply src/components/SignInButton/type.ts]: yes, it works

File: src/redux/saga/checkFirebaseUsersSaga.ts (hunk 16 above)
Comment (AndrewSakovich) [reply src/redux/saga/checkFirebaseUsersSaga.ts]: already done

**Review by SHEMPEL133 — APPROVED** [review]
